segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -222,10 +221,302 @@ def generate_test_label_map(size_img=(1024, 1024), num_A=100, num_B=100, num_C=1
+
+    if seed:
+        np.random.seed(seed)
+
+    # generate some jitter for each x and y coordinate
+    # the way we do this is to generate random offset based on the jitter_factor multiplier
+    # and sampling from a uniform distribution with an additional jitter_factor and jitter_multiplier multiplier
+    ring_region_1_x_jitter = (ring_region_1[0] + np.around(np.random.uniform(-jitter_factor * jitter_multiplier, jitter_factor * jitter_multiplier, len(ring_region_1[0])))).astype(np.int16)
+    ring_region_1_y_jitter = (ring_region_1[1] + np.around(np.random.uniform(-jitter_factor * jitter_multiplier, jitter_factor * jitter_multiplier, len(ring_region_1[1])))).astype(np.int16)
+    ring_region_2_x_jitter = (ring_region_2[0] + np.around(np.random.uniform(-jitter_factor * jitter_multiplier, jitter_factor * jitter_multiplier, len(ring_region_2[0])))).astype(np.int16)
+    ring_region_2_y_jitter = (ring_region_2[1] + np.around(np.random.uniform(-jitter_factor * jitter_multiplier, jitter_factor * jitter_multiplier, len(ring_region_2[1])))).astype(np.int16)
+
+    # now ensure the jitter coordinates don't fall out of bounds
+    # this we need to do because the offsets could fall out of bounds
+    ring_region_1_jitter_indices = (ring_region_1_x_jitter >= 0) & (ring_region_1_x_jitter < size_img[0]) & \
+                                   (ring_region_1_y_jitter >= 0) & (ring_region_1_y_jitter < size_img[1])
+
+    ring_region_2_jitter_indices = (ring_region_2_x_jitter >= 0) & (ring_region_2_x_jitter < size_img[0]) & \
+                                   (ring_region_2_y_jitter >= 0) & (ring_region_2_y_jitter < size_img[1])
+
+    ring_region_1_x_jitter_coords = ring_region_1_x_jitter[ring_region_1_jitter_indices]
+    ring_region_1_y_jitter_coords = ring_region_1_y_jitter[ring_region_1_jitter_indices]
+    ring_region_2_x_jitter_coords = ring_region_2_x_jitter[ring_region_2_jitter_indices]
+    ring_region_2_y_jitter_coords = ring_region_2_y_jitter[ring_region_2_jitter_indices]
+
+    # add an additional random point removal for clearer results
+    # might be a bit overkill but it definitely did help
+    # over trying to fine tune jitter factor and what not
+    # number of points to keep is inversely proportional to
+    # how far away the ring is from the nucleus or membrane respectively
+    ring_region_1_selection = np.random.choice(len(ring_region_1_x_jitter_coords), int(len(ring_region_1_x_jitter_coords) / jitter_multiplier))
+    ring_region_2_selection = np.random.choice(len(ring_region_2_x_jitter_coords), int(len(ring_region_2_x_jitter_coords) / jitter_multiplier))
+
+    ring_region_1_x_jitter_coords = ring_region_1_x_jitter_coords[ring_region_1_selection]
+    ring_region_1_y_jitter_coords = ring_region_1_y_jitter_coords[ring_region_1_selection]
+    ring_region_2_x_jitter_coords = ring_region_2_x_jitter_coords[ring_region_2_selection]
+    ring_region_2_y_jitter_coords = ring_region_2_y_jitter_coords[ring_region_2_selection]
```

Comment:
since `jitter_factor` wasn't included in this logic, it made `jitter_factor=0` in higher level functions not work properly.  If we remove partitioning, this can be removed as well.